File: 21553 - Sai Nandan D/Week 8/162_Find Peak Element.py
# Approach 1: Recursive
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        def find_peak(l, r):
            if l == r: return l
            mid = (l + r) // 2
            if nums[mid] > nums[mid+1]: r = mid
            else: l = mid + 1
            return find_peak(l, r)
        
        return find_peak(0, len(nums)-1)
      
      
# Approach 2: Iterative
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        l, r = 0, len(nums)-1
        while l < r:
            mid = (l+r) // 2
            if nums[mid] < nums[mid+1]: l = mid+1
            else: r = mid                
        return l

File: 21561 - Suvam Tamang/Week 8/162_Find Peak Element.java
class Solution {
    public int findPeakElement(int[] nums) {
        int low = 0, high = nums.length-1, mid;
        while(low < high){
            mid = (low + high) / 2;
            if(nums[mid] > nums[mid + 1])
                high = mid;
            else 
                low = mid + 1;
        }
        return low;
    }
}
